### PR TITLE
ecmult: move `_ecmult_odd_multiples_table_globalz_windowa`

### LINKED ISSUE #1035: Move secp256k1_ecmult_odd_multiples_table_globalz_windowa and fix docs
`secp256k1_ecmult_odd_multiples_table_globalz_windowa` is in the wrong module.  This function had been used within ecmult in the past but since 8c1c831bdb083dfa8b50fac16a0e17a7e1df4064, it is only used in ecmult_const. This also means its comment is wrong.

_Originally posted by @real-or-random in https://github.com/bitcoin-core/secp256k1/issues/942#issuecomment-987765042_

### PR DESCRIPTION
**Pull request by siv2r**
Fixes #1035

**Changes:**
    - move `secp256k1_ecmult_odd_multiples_table_globalz_windowa` function from ecmult to ecmult_const
    - remove outdated comment

### CODE REVIEW

**Comment by real-or-random** [comment]
Thanks! Can you rebase though? :P

**Comment by siv2r** [comment]
> Can you rebase though?

Done :)

**Review by real-or-random — APPROVED** [review]
utACK 6287fa8c43736a9e187862a1ad0f7d159037e3fd

**Comment by robot-dreams** [comment]
Concept ACK, thanks for addressing this!

Though should this wait for #899 to merge first to avoid a merge conflict in that one (which is a bit longer and already has two-ish ACKs)?

**Comment by real-or-random** [comment]
> Though should this wait for #899 to merge first to avoid a merge conflict in that one (which is a bit longer and already has two-ish ACKs)?

Oh, thanks for pointing this out.

File: src/ecmult_const_impl.h [hunk 1]
```diff
@@ -12,6 +12,22 @@
 #include "ecmult_const.h"
 #include "ecmult_impl.h"
 
+/** Fill a table 'pre' with precomputed odd multiples of a.
+ *
+ *  The resulting point set is brought to a single constant Z denominator, stores the X and Y
+ *  coordinates as ge_storage points in pre, and stores the global Z in rz.
```
Comment (real-or-random) [inline src/ecmult_const_impl.h:18]:
```suggestion
 *  coordinates as ge_storage points in pre, and stores the global Z in globalz.
```

Who would have expected that the global Z is stored in `globalz`?

**Comment by real-or-random** [comment]
roconnor-blockstream told me that he doesn't mind rebasing #899 . 

By the way, it's debatable if this function should be prefixed with `ecmult` or `ecmult_const`... But I don't know, the enitre ecmult_const module is not really separated from the ecmult module (reuses `WINDOW_A` and the like). I believe this could be made cleaner but that could be done in a separate PR (see also #1039).

**Comment by robot-dreams** [comment]
utACK 6287fa8c43736a9e187862a1ad0f7d159037e3fd (aside from @real-or-random 's suggested comment fixup)

I agree that:
- `secp256k1_ecmult_odd_multiples_table_globalz_windowa` is only called from `ecmult_const_impl.h` and thus should be moved there
- The function is no longer used for computation of `a*P + b*G`, thus the comment should be removed

**Comment by siv2r** [comment]
Included the suggestion of https://github.com/bitcoin-core/secp256k1/pull/1053#discussion_r774774829

**Review by real-or-random — APPROVED** [review]
utACK 05e049b73c69002f498c3c9c21555fd91f95ccac

**Comment by robot-dreams** [comment]
utACK 05e049b73c69002f498c3c9c21555fd91f95ccac (`diff` between removed and added lines is exactly as expected)

**Pull request merged**